packages/edit-post/src/components/sidebar/template/actions.js

```diff
@@ -53,6 +59,43 @@ function PostTemplateActions() {
 	return (
 		<>
 			<div className="edit-post-template__actions">
+				{ !! ( template && template.wp_id ) && (
+					<Button
+						isLink
+						isDestructive
+						onClick={ () => {
+							if (
+								// eslint-disable-next-line no-alert
+								window.confirm(
+									__(
+										'Are you sure you want to delete this template? It may be currently in use by other pages or posts.'
+									)
+								)
+							) {
+								editPost( {
+									template: '',
+								} );
+								const newAvailableTemplates = pickBy(
+									editorSettings.availableTemplates,
+									( _title, id ) => {
+										return id !== template.slug;
+									}
+								);
+								updateEditorSettings( {
+									...editorSettings,
+									availableTemplates: newAvailableTemplates,
```

Comment:
When creating new templates (new link there), we don't update this setting at the moment, it seems we should have that covered similarly. I think ideally, this should be a REST API request but right now patching the setting might be ok.

Also, I noticed that `editorSettitngs`  is only used in the event handler which means we don't need to return it in useSelect, we can just use the selector with the new syntax


```
const { getSettings ) = useSelect( 'core/block-editor' );
```

Reply:
>his should be a REST API request but right now patching the setting might be ok.

Right now there are filters in the core that allow plugins and themes to select which templates are available to select by the user on a post type or even a specific post. We could not just show make a rest request on the templates post type. I think we would need a specific endpoint that returns templates available for a given post where the existing core filters are executed.